src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SwaggerSamClient.scala

```diff
@@ -0,0 +1,128 @@
+package org.broadinstitute.dsde.workbench.leonardo.auth
+
+import java.io.{ByteArrayInputStream, File}
+import java.util.concurrent.TimeUnit
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
+import com.google.api.client.json.jackson2.JacksonFactory
+import com.google.api.services.plus.PlusScopes
+import com.google.auth.oauth2.ServiceAccountCredentials
+import com.google.common.cache.{CacheBuilder, CacheLoader}
+import com.google.gson.Gson
+import com.google.gson.internal.LinkedTreeMap
+import io.swagger.client.ApiClient
+import io.swagger.client.api.{GoogleApi, ResourcesApi}
+import org.broadinstitute.dsde.workbench.leonardo.model.ClusterName
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+
+case class UserEmailAndProject(userEmail: WorkbenchEmail, googleProject: GoogleProject)
+
+class SwaggerSamClient(samBasePath: String, cacheExpiryTime: Int, cacheMaxSize: Int,  leoEmail: WorkbenchEmail, leoPem: File) {
```

Comment:
For future housekeeping we should `PetServiceAccountProvider` use `SwaggerSamClient`